conformance/utils/echo/pod.go

```diff
@@ -113,10 +113,16 @@ func compareRequest(exp http.ExpectedResponse, resp Response) error {
 	return nil
 }
 
-// copied from conformance/http to get the ipv6 matching
-func calculateHost(reqHost, scheme string) string {
-	host, port, err := net.SplitHostPort(reqHost)
+func calculateHost(t *testing.T, reqHost, scheme string) string {
```

Comment:
I realize that this PR is just following the already established pattern, but why do we need two copies of the same calculateHost function? Can't pod.go just use the one in http.go instead of duplicating the code?

Reply:
Unexported functions in different packages I'm afraid. I could add it to a general utils package if that'll be helpful?

Reply:
I think a shared package is a good idea here so we don't introduce accidental variations between packages. A couple ideas:

1. Introduce a new `conformance/utils/net` package that could house this and `ipv6SafeHost` functions
2. Export `calculateHost` from `conformance/utils/http` function since it's already there.

Since this package already depends on `conformance/utils/http`, I think I'd lean towards 2.

Reply:
> Since this package already depends on conformance/utils/http, I think I'd lean towards 2.

+1

Reply:
Nice catch; I missed the existing dependency. Done!